### PR TITLE
Import db turbo mode

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added extra option in import-db mode that can skip headers signature checks
- implemented disabled crypto components (suite, scalar, point, single signer & multi signer)
- disabled heartbeat sender during the import-db process

### CODE REVIEW

File: cmd/node/main.go [hunk 1]
```diff
@@ -744,6 +751,7 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 		KeyGen:                               cryptoParams.KeyGenerator,
 		PrivKey:                              cryptoParams.PrivateKey,
 		ActivateBLSPubKeyMessageVerification: systemSCConfig.StakingSystemSCConfig.ActivateBLSPubKeyMessageVerification,
+		UseMockSigVerifier:                   importDbNoSigCheckFlag,
```
Comment (sasurobert) [inline cmd/node/main.go:754]: UseDisabledSigVerifier

File: crypto/signing/disabled/singlesig/disabled.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package singlesig
+
+import "github.com/ElrondNetwork/elrond-go/crypto"
+
+const signature = "signature"
+
+// Disabled represents a disabled singlesigner implementation
+type Disabled struct {
```
Comment (sasurobert) [inline crypto/signing/disabled/singlesig/disabled.go:8]: Disabled what ?

File: factory/cryptoComponents.go [hunk 3]
```diff
@@ -9,6 +9,9 @@ import (
 	"github.com/ElrondNetwork/elrond-go/crypto"
 	"github.com/ElrondNetwork/elrond-go/crypto/peerSignatureHandler"
 	"github.com/ElrondNetwork/elrond-go/crypto/signing"
+	disabledCrypto "github.com/ElrondNetwork/elrond-go/crypto/signing/disabled"
+	disabledmultisig "github.com/ElrondNetwork/elrond-go/crypto/signing/disabled/multisig"
```
Comment (sasurobert) [inline factory/cryptoComponents.go:13]: disabledMultiSig

File: factory/cryptoComponents.go [hunk 4]
```diff
@@ -9,6 +9,9 @@ import (
...
+	disabledsig "github.com/ElrondNetwork/elrond-go/crypto/signing/disabled/singlesig"
```
Comment (sasurobert) [inline factory/cryptoComponents.go:14]: disabledSingleSig

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: crypto/signing/disabled/singlesig/disabled.go (hunk 2 above)
Comment (iulianpascalau) [reply crypto/signing/disabled/singlesig/disabled.go:8]: singlesig.Disabled, I think it's pretty obvious

File: cmd/node/main.go (hunk 1 above)
Comment (iulianpascalau) [reply cmd/node/main.go:754]: :1st_place_medal:

Comment (iulianpascalau) [reply cmd/node/main.go:755]: done

File: factory/cryptoComponents.go (hunk 3 above)
Comment (iulianpascalau) [reply factory/cryptoComponents.go:13]: done

File: factory/cryptoComponents.go (hunk 4 above)
Comment (iulianpascalau) [reply factory/cryptoComponents.go:14]: done

File: crypto/signing/disabled/multisig/disabled_test.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+func TestDisabled_MethodsShouldNotPanic(t *testing.T) {
+	defer func() {
+		r := recover()
+		if r != nil {
+			assert.Fail(t, fmt.Sprintf("should have not panic: %v", r))
+		}
+	}()
+
+	d := &Disabled{}
+	recovInstance, err := d.Create(nil, 0)
```
Comment (SebastianMarian) [inline crypto/signing/disabled/multisig/disabled_test.go:20]: recov comes from recovered ?

File: crypto/signing/disabled/point.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+func (dp *disabledPoint) MarshalBinary() ([]byte, error) {
+	return []byte(marshaledPoint), nil
+}
+
+// UnmarshalBinary returns nil
+func (dp *disabledPoint) UnmarshalBinary(_ []byte) error {
+	return nil
+}
+
+// Equal return false
```
Comment (SebastianMarian) [inline crypto/signing/disabled/point.go:20]: returns

File: crypto/signing/disabled/singlesig/disabled.go (hunk 2 above)
Comment (SebastianMarian) [reply crypto/signing/disabled/singlesig/disabled.go:8]: DisabledSingleSig ? to keep the same pattern used above in crypto/signing/disabled/point.go -> DisabledPoint
and I think you can rename this file to singlesig.go and move it in crypto/signing/disabled/ folder alongside point.go ?

File: crypto/signing/disabled/suite.go [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+const pointLen = 96
+
+type disabledSuite struct{}
+
+// NewDisabledSuite creates a new disabled suite
+func NewDisabledSuite() *disabledSuite {
+	return &disabledSuite{}
+}
+
+// returns the disabled string
```
Comment (SebastianMarian) [inline crypto/signing/disabled/suite.go:21]: // String ...

File: crypto/signing/disabled/suite.go [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+func (ds *disabledSuite) CreateScalar() crypto.Scalar {
+	return &disabledScalar{}
+}
+
+// PointLen returns 0
+func (ds *disabledSuite) PointLen() int {
+	return pointLen
+}
+
+// CreatePoint create a disabledPoint instance
```
Comment (SebastianMarian) [inline crypto/signing/disabled/suite.go:41]: creates

File: crypto/signing/disabled/suite.go [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+func (ds *disabledSuite) CreatePoint() crypto.Point {
+	return &disabledPoint{}
+}
+
+// CreatePointForScalar create a disabledPoint instance
```
Comment (SebastianMarian) [inline crypto/signing/disabled/suite.go:46]: creates

File: crypto/signing/disabled/multisig/disabled_test.go (hunk 5 above)
Comment (iulianpascalau) [reply crypto/signing/disabled/multisig/disabled_test.go:20]: yes

Comment (iulianpascalau) [reply crypto/signing/disabled/multisig/disabled_test.go:20]: changed

File: crypto/signing/disabled/point.go (hunk 6 above)
Comment (iulianpascalau) [reply crypto/signing/disabled/point.go:20]: done

File: crypto/signing/disabled/suite.go (hunk 7 above)
Comment (iulianpascalau) [reply crypto/signing/disabled/suite.go:21]: done

File: crypto/signing/disabled/suite.go (hunk 8 above)
Comment (iulianpascalau) [reply crypto/signing/disabled/suite.go:41]: done

File: crypto/signing/disabled/suite.go (hunk 9 above)
Comment (iulianpascalau) [reply crypto/signing/disabled/suite.go:46]: done

File: crypto/signing/disabled/singlesig/disabled.go (hunk 2 above)
Comment (iulianpascalau) [reply crypto/signing/disabled/singlesig/disabled.go:8]: done, renamed. I would keep them as they are as to comply with the rest of the crypto package implementations.

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by LucianMincu — COMMENTED** [review]
👍

**Review by LucianMincu — APPROVED** [review]

**Pull request merged**